src/site/markdown/in_depth_walkthrough.md

```diff
@@ -0,0 +1,316 @@
+the features produced by a [CompiledSemantics](aloha-core/scaladocs/index.html#com.eharmony.aloha.semantics.compiled.CompiledSemantics) instance 
+parameterized by an [CompiledSemanticsCsvPlugin](aloha-core/scaladocs/index.html#com.eharmony.aloha.semantics.compiled.plugin.csv.CompiledSemanticsCsvPlugin) operate 
+on instances of [CsvLine](aloha-core/scaladocs/index.html#com.eharmony.aloha.semantics.compiled.plugin.csv.CsvLine).  It 
+is therefore necessary to produce instances of CsvLine to make use of this plugin.  This is pretty easy to do though.
+
+```scala
+val csvLines = CsvLines(
+  Map(                     // 0-based field indices
+    "height" -> 0, 
+    "weight" -> 1
+  ),
+  Map.empty,               // enums: Map[String, Enum] 
+  "\t",                    // fs (field separator) 
+  ",",                     // ifs (intra-field separator, for vectorized inputs)
+  (s: String) => s == "",  // missingData function 
+  false,                   // errorOnOptMissingField 
+  false                    // errorOnOptMissingEnum
+)
+
+val csvLinesWithDefaults = CsvLines(Map("height" -> 0, "weight" -> 1))
+
+require(csvLinesWithDefaults == csvLines)  // This requirement is satisfied.
+```
+
+Once in possession of a CsvLines instance, it's easy to generate CsvLine instances:  
+
+```scala
+val oneCsvLine: CsvLineImpl = csvLines("72, 175") // height = 72 inches, weight = 175 lbs.
+val manyCsvLinesUsingVarArgs: Vector[CsvLineImpl] = csvLines("72, 175", "70, 180", "64, 110")
+val csvLineIterator: Iterator[CsvLineImpl] = csvLines(Iterator("72, 175", "70, 180", "64, 110"))
+val csvLineVector: Seq[CsvLineImpl] = csvLines(Seq("72, 175", "70, 180", "64, 110"))
+val nonStrictVectorOfCsvLine = csvLines.nonStrict(Vector("72, 175", "70, 180", "64, 110"))
+```
+
+## Factory
+
+Factories are important to create models: 
+
+```scala
+// Scala Code
+import com.eharmony.aloha.factory.ModelFactory
+import com.eharmony.aloha.semantics.compiled.plugin.csv.CsvTypes.CsvType._
+
+// vvvvv  Define the semantics  vvvvvvvvvvvvvvvvvvvvvv
+
+// -----  Define the plugin  -------------------------
+val plugin = CompiledSemanticsCsvPlugin(
+  "height" -> IntType
+  "weight" -> FloatOptionType
+)
+
+val imports = Seq("scala.math._")
+val cacheDir: java.io.File = getCacheDir() // Get a place to cache features.
+val semantics = CompiledSemantics(TwitterEvalCompiler(classCacheDir = Option(cacheDir), plugin, imports)
+// ^^^^^  Define the semantics  ^^^^^^^^^^^^^^^^^^^^^^
+
+// Get a typed factory.
+val untypedFactory = ModelFactory.defaultFactory
+
+val typedFactory = untypedFactory.toTypedFactory[CsvLine, Double](semantics)
+
+val modelFile: java.io.File = ...
+
+// type: scala.util.Try[com.eharmony.aloha.models.Model[CsvLine, Double]]
+val modelAttempt = typedFactory.fromFile(modelFile)
+val model = modelAttempt.get 
+```
```

Comment:
Add a final section encouraging people to commit their own ideas as well for other formats and other model types.